data-contracts/README.md

```diff
@@ -0,0 +1,439 @@
+
+Let's also implement the `BaseConsumer` class, configured to use this `major_version` of the schema:
+
+```kotlin
+class MembershipConsumer: BaseConsumer<String, Membership>(mapOf(
+    ConsumerConfig.GROUP_ID_CONFIG to "app-schema-v1",
+    ConsumerConfig.AUTO_OFFSET_RESET_CONFIG to "earliest",
+    ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG to "org.apache.kafka.common.serialization.StringDeserializer",
+    ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG to "io.confluent.kafka.serializers.KafkaAvroDeserializer",
+    AbstractKafkaSchemaSerDeConfig.LATEST_COMPATIBILITY_STRICT to true,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_VERSION to false,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_WITH_METADATA to "major_version=1"
+)) {
+
+    override fun consumeRecord(
+        key: String,
+        value: Membership
+    ) {
+        logger.info("Received Membership ${key}, ${value}")
+    }
+}
+```
+
+The `consumeRecord` function is where we would typically start the business logic of actually "consuming" Kafka events. This implementation simply logs the consumed records to the provided `Logger` instance in the superclass.
+
+To exercise these producer and consumer implementations, we created a `main` function in the `ApplicationMain` class to start a consumer instance and a producer instance periodically send random events to the `membership-avro` topic.
+
+<details>
+    <summary>Main function</summary>
+
+```kotlin
+@JvmStatic
+fun main(args: Array<String>) {
+    runBlocking {
+        println("Starting application main...")
+        println(args.joinToString(" "))
+        
+        val messageInterval = 1.toDuration(DurationUnit.SECONDS)
+        val sendDuration = 100.toDuration(DurationUnit.SECONDS)
+
+        val producer = MembershipProducer()
+        val consumer = MembershipConsumer()
+
+        // start a thread with a consumer instance
+        thread {
+            consumer.start(listOf("membership-avro"))
+        }
+
+        // every 1 second for the next  100 seconds, send a randomly-generated event to the kafka topic
+        coroutineScope {
+            launch {
+                val until = Clock.System.now().plus(sendDuration)
+                while(Clock.System.now().compareTo(until) < 0) {
+                    val userId = UUID.randomUUID().toString()
+                    val membership = Membership.newBuilder()
+                        .setUserId(userId)
+                        .setStartDate(LocalDate.now().minusDays(Random.nextLong(100, 1000)))
+                        .setEndDate(LocalDate.now().plusWeeks(Random.nextLong(1, 52)))
+                        .build()
+                    producer.send("membership-avro", userId, membership)
+                    delay(messageInterval.inWholeSeconds)
+                }
+            }
+        }
+        producer.close()
+    }
+}
+```
+
+</details>
+
+Running this application will print the events being consumed from Kafka:
```

Comment:
Is there a gradle command to run it? or instruction to run from IntelliJ? I tried opening the `data-contracts` directory in intellij but it's not finding the `shared` module:
![image](https://github.com/user-attachments/assets/556a9d0f-ae8a-487f-9f3d-1ed102806750)
I think that `settings.gradle.kts` need some `include` directives to avoid these errors, but wondering if there is something different about your IntelliJ setup that avoids them